client/src/components/Signup/Signup.js

```diff
@@ -0,0 +1,181 @@
+            validated: false,
+            name: '',
+            email: '',
+            password: '',
+            matchingPassword: '',
+            errors: {
+                name: '',
+                email: '',
+                password: '',
+                matchingPassword: ''
+            }
+        };
+    }
+
+    validateForm(errors) {
+        let valid = true;
+        for (let name of ['name', 'email', 'password', 'matchingPassword']) {
+            this.handleChange({
+                target: {
+                    name,
+                    value: this.state[name]
+                }
+            });
+        }
+        for (let message of Object.values(errors)) {
+            if (message.length > 0) valid = false;
+        }
+        this.setState({validated: true});
+        return valid;
+    }
+
+    handleSubmit(event) {
+        const { onSignup } = this.props;
+        const { errors, email, password, matchingPassword, name } = this.state;
+        event.preventDefault();
+        if (this.validateForm(errors)) {
+            onSignup({email, password, matchingPassword, name});
+        }
+    }
+
+    handleChange(event) {
+        const { errors, password } = this.state;
+        const { name, value } = event.target;
+        if (event.preventDefault) event.preventDefault();
+
+        switch (name) {
+            case 'name':
+                errors.name = value
+                    ? ''
+                    : 'Full Name is required';
+                break;
+            case 'email':
+                errors.email = validEmailRegex.test(value)
+                    ? ''
+                    : 'Email is not valid';
+                break;
+            case 'password':
+                errors.password = value.length < 8
+                    ? 'Password must be 8 characters long'
+                    : '';
+                break;
+            case 'matchingPassword':
+                errors.matchingPassword = value === password
+                    ? ''
+                    : 'Passwords must match';
+                break;
+            default:
+                break;
+        }
+
+        this.setState({errors, [name]: value})
+    }
+
+    render() {
+        const { email, password, matchingPassword, name, errors, validated } = this.state;
+        const { error, result } = this.props;
+
+        return (
+            <Card className="signup">
+                <Card.Header>
+                    <Card.Title className="text-center">
+                        <Logo size="35px"/>
+                        <span>
+                            Sign Up
+                        </span>
+                    </Card.Title>
+                </Card.Header>
+                <Card.Body>
+                    {!result &&
+                        <Form className="signup-form" noValidate onSubmit={event => this.handleSubmit(event)}>
+                            <Form.Group>
+                                <Form.Label>Full Name</Form.Label>
```

Comment:
Is it more user friendly to have a First Name and a Last Name field instead of a single one?

Reply:
I started with that, and it might make sense to because usually in emails you just want to address by first name, so I'll add that back in